odoo/osv/expression.py

```diff
@@ -1342,83 +1388,89 @@ def push_result(query, params):
                             right = datetime.combine(right, time.min)
                         push((left, operator, right), model, alias)
                     else:
-                        expr, params = self.__leaf_to_sql(leaf, model, alias)
-                        push_result(expr, params)
+                        push_result(self.__leaf_to_sql(leaf, model, alias))
 
                 elif field.translate and isinstance(right, str) and left == field.name:
-                    sql_operator = {'=like': 'like', '=ilike': 'ilike'}.get(operator, operator)
-                    expr = ''
-                    params = []
+                    model_raw_trans = model.with_context(prefetch_langs=True)
+                    sql_field = model_raw_trans._field_to_sql(alias, field.name, self.query)
+                    sql_operator = SQL_OPERATORS[operator]
+                    sql_exprs = []
 
                     need_wildcard = operator in ('like', 'ilike', 'not like', 'not ilike')
                     if not need_wildcard:
                         right = field.convert_to_column(right, model, validate=False).adapted['en_US']
 
-                    if (need_wildcard and not right) or (right and sql_operator in NEGATIVE_TERM_OPERATORS):
-                        expr += f'"{alias}"."{field.name}" is NULL OR '
+                    if (need_wildcard and not right) or (right and operator in NEGATIVE_TERM_OPERATORS):
+                        sql_exprs.append(SQL("%s IS NULL OR", sql_field))
```

Comment:
Should we implement `__iadd__`/`__add__` on SQL ? Or you prefer force people to use join instead ?

Reply:
This use-case is actually crappy. It should properly compose with what comes on its right. Adding `__add__` will encourage crappy SQL coding, which I want to avoid.